storage_service/common/which.py

```diff
@@ -0,0 +1,70 @@
+"""Backport of ``shutil.which``.
+
+Extracted from ``Lib/shutil.py`` (Python 3.7).
```

Comment:
Did you do any changes to make it work in Python 2.7? I'd be nice to point them out in here if that's the case.

Reply:
To my surprise, it worked out of the box. I've just added their tests too, which needed some tweaking but they're working!